### PR TITLE
[jaxrs-cxf][jaxrs-spec] Add missing @NotNull annotation for required body parameters when using BeanValidation, fixes #4280

### LINKED ISSUE #4280: [BUG] NotNull annotation is not set for requiered body in jaxrs-cxf
#### Bug Report Checklist

- [X ] Have you provided a full/minimal spec to reproduce the issue?
- [ ] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [X ] What's the version of OpenAPI Generator used?
- [X ] Have you search for related issues/PRs?
- [X ] What's the actual output vs expected output?
- [ ] [Optional] Bounty to sponsor the fix ([example](https://www.bountysource.com/issues/66123212-javascript-client-produces-a-wrong-object-for-a-string-enum-type-that-is-used-with-ref))

##### Description

When validation is activated with jaxrs-cxf and a body is requiered the annotation @NotNull is not set

##### openapi-generator version

4.1.3

##### OpenAPI declaration file content or url

<configuration>

              <language>jaxrs-cxf</language>
              <generateApis>true</generateApis>
              <generateApiTests>false</generateApiTests>
              <generateModels>true</generateModels>
              <generateSupportingFiles>false</generateSupportingFiles>
              <addCompileSourceRoot>true</addCompileSourceRoot>
              <configOptions>
                <serializableModel>true</serializableModel>
                <sourceFolder>src/main/java</sourceFolder>
                <dateLibrary>java8</dateLibrary><java8>true</java8>
                <invokerPackage>${package-path}.invoker</invokerPackage>
                <modelPackage>${package-path}.model</modelPackage>
                <apiPackage>${package-path}.api</apiPackage>
                <serializableModel>true</serializableModel>
                <useBeanValidation>true</useBeanValidation>
                <useReflectionEqualsHashCode>true</useReflectionEqualsHashCode>
                <useGenericResponse>true</useGenericResponse>
                <booleanGetterPrefix>get</booleanGetterPrefix>
                <addConsumesProducesJson>false</addConsumesProducesJson> 
                <useAnnotatedBasePath>true</useAnnotatedBasePath>               
              </configOptions>
            </configuration>

##### Suggest a fix

You will find the new mustasche for bodyParams

[bodyParams.zip](https://github.com/OpenAPITools/openapi-generator/files/3774553/bodyParams.zip)

### PR DESCRIPTION
**Pull request by ruedigerk**
This PR adds missing @NotNull annotations for required body parameters when using BeanValidation and fixes #4280.

I fixed all jaxrs generators which use identical bodyParams.mustache files (my company is affected by this bug in the jaxrs-spec generator):
- jaxrs-cxf
- jaxrs-cxf-extended
- jaxrs-spec

Please review by Java technical committee: 
@bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10) @bkabrda (2020/01)

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [x] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/JavaJaxRS/spec/bodyParams.mustache [hunk 1]
```diff
@@ -1 +1 @@
-{{#isBodyParam}}{{#useBeanValidation}}@Valid {{/useBeanValidation}}{{{dataType}}} {{paramName}}{{/isBodyParam}}
\ No newline at end of file
+{{#isBodyParam}}{{#useBeanValidation}}@Valid {{#required}}@NotNull {{/required}}{{/useBeanValidation}}{{{dataType}}} {{paramName}}{{/isBodyParam}}
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/JavaJaxRS/spec/bodyParams.mustache:1]: @ruedigerk thanks for the PR. I think we need the `{{^isNulalble}} ... {{/isNullable}}` check as well to ensure the required parameter is not nullable.

Comment (ruedigerk) [reply modules/openapi-generator/src/main/resources/JavaJaxRS/spec/bodyParams.mustache:1]: After thinking about this, I think adding the NotNull annotation for the "not nullable" case is not correct. If the parameter type is not nullable, it is not allowed to pass the value "null" as the parameter value, as far as I understand the spec. But I think it is still allowed to omit the parameter completely (if it is not required). 

So omitting a non-required parameter that is non-nullable still (correctly) results in a null value on the Java side, therefore the NotNull annotation must not be added in this case. Therefore checking only for "required" seems to be the right thing to do to me.

Comment (ruedigerk) [reply modules/openapi-generator/src/main/resources/JavaJaxRS/spec/bodyParams.mustache:1]: But following my own reasoning, the NotNull annotation should no be added if the parameter is required but nullable. Ok, I think I misunderstood your comment and that was what you meant all along. I will change it accordingly.

Comment (ruedigerk) [reply modules/openapi-generator/src/main/resources/JavaJaxRS/spec/bodyParams.mustache]: Hello @wing328, I fixed the issue you found. Please review again.

**Comment by ruedigerk** [comment]
I have resolved all issues found in the code review and the PR is waiting since 8 months to be merged. Is there a problem that still needs to be addressed?

**Comment by ruedigerk** [comment]
@wing328 please merge this PR. I have completed all tasks and I am waiting since months for this fix to be merged. I just rebased it on top of the latest master branch an reran all checks.

**Comment by wing328** [comment]
@ruedigerk sorry that I missed the update :( Too many users tagged me

Thanks for taking my feedback and update the PR accordingly. It looks good to me.

**Review by wing328 — APPROVED** [review]

**Comment by wing328** [comment]
@ruedigerk thanks for the PR, which has been included in the v5.1.0 release: https://twitter.com/oas_generator/status/1373636654024380423

**Pull request merged**